### ISSUE TITLE
"no local sources enabled" randomly during parallel solves

### ISSUE THREAD

**Issue by coryb** [openllb/hlb#152]
Using this HLB we will compile the hlb tool in parallel for Windows, Darwin and Linux:

```
import go from fs {
    image "openllb/go.hlb"
}

group default() {
	go.buildCommon fs {
		local "." with option {
			includePatterns "**/*.go" "go.mod" "go.sum"
		}
	} "github.com/openllb/hlb/cmd/hlb"
}
 ```
However I see the "no local sources enabled" error around 3/4 times:
```
$ ./hlb run ./test.hlb
[+] Building 2.1s (4/6)
 => CACHED [import go] docker-image://docker.io/openllb/go.hlb:latest                                   0.0s
 => => resolve docker.io/openllb/go.hlb:latest                                                          1.3s
 => compiling [default]                                                                                 2.0s
 => CANCELED docker-image://docker.elastic.co/beats-dev/golang-crossbuild:1.14.2-main                   0.0s
 => => resolve docker.elastic.co/beats-dev/golang-crossbuild:1.14.2-main                                0.0s
 => ERROR local://sha256:ecc4116a1fa41562e2426429f2e4b9d711f9ce72481830e244e9e0d66a32325b (**/*.go)     0.0s
 => => transferring sha256:ecc4116a1fa41562e2426429f2e4b9d711f9ce72481830e244e9e0d66a32325b:            0.0s
------
 > local://sha256:ecc4116a1fa41562e2426429f2e4b9d711f9ce72481830e244e9e0d66a32325b (**/*.go):
------
failed to solve: rpc error: code = Unknown desc = no local sources enabled
```

If I keep retrying it does eventually work.  There is clearly a race condition and probably shared state between parallel solves, but I have not had any luck finding it yet.

**Comment by hinshun** [comment]
I thought this might be related to: https://github.com/moby/buildkit/pull/2585
But I looked over the [git blame](https://github.com/moby/buildkit/commit/2e9987ad1664cd89c04deaa528a7e56111c0c378#diff-2ef0a0b20ac6c4fb60c7b0e1c8c2d686c929f4c2fae387a5e268cc0e6e874517R96-R104) and it was only introduced in July 2020, after your report in May 2020.

**Comment by coryb** [comment]
I dont recall seeing this error in a while, although I have not tried this specific build in a while.  Perhaps it has been fixed.

**Comment by hinshun** [comment]
Closing for now. We can open a new issue if we see it again.
